cli/azd/pkg/infra/provisioning/provider.go

```diff
@@ -62,17 +63,26 @@ var (
 )
 
 const (
+	// Provider Provider Types
+
 	Bicep     ProviderKind = "bicep"
 	Arm       ProviderKind = "arm"
 	Terraform ProviderKind = "terraform"
 	Pulumi    ProviderKind = "pulumi"
 	Test      ProviderKind = "test"
+
+	// Provision Scope Types
+
+	SubscriptionScope  ScopeKind = "subscription"
+	ResourceGroupScope ScopeKind = "resourcegroup"
+	NotSpecifiedScope  ScopeKind = ""
 )
 
 type Options struct {
 	Provider ProviderKind `yaml:"provider"`
 	Path     string       `yaml:"path"`
 	Module   string       `yaml:"module"`
+	Scope    ScopeKind    `yaml:"scope"`
```

Comment:
I think we should be inferring the type of scope from the bicep file itself (we can detect this by looking at the schema of the compiled ARM template).  That way we won't open ourselves up to mismatches between what azure.yaml says and what the actual template we are trying to deploy is.